assets/js/src/form_editor/store/controls.jsx

```diff
@@ -6,9 +6,14 @@ export default {
     if (select('mailpoet-form-editor').getIsFormSaving()) {
       return;
     }
-    dispatch('mailpoet-form-editor').saveFormStarted();
-    const formData = select('mailpoet-form-editor').getFormData();
     dispatch('mailpoet-form-editor').removeNotice('save-form');
+    const formData = select('mailpoet-form-editor').getFormData();
+    if (!formData.settings.segments || formData.settings.segments.length === 0) {
```

Comment:
I am not sure about this one. The responsibility of this control is not to validate, and so I think this should be moved to some other place. I am not sure if or where, but I think that validation should be done elsewhere.